### PR TITLE
Add Deno socket server support

### PR DESCRIPTION
**Pull request by tim-smart**
Closes EFF-149

## Summary
- add native Deno TCP, Unix, and TLS socket server constructors and layers
- supervise connection handlers and apply resilient accept-loop backoff
- cover TCP/Unix integration and accept-loop edge cases

## Testing
- `nix develop -c deno task test --run packages/platform-deno/test/DenoSocketServer.test.ts`
- `nix develop -c deno check .`
- `nix develop -c pnpm check`
- `nix develop -c pnpm lint`

## Summary by CodeRabbit

* **New Features**
  * Added native Deno `SocketServer` support for TCP, Unix, and TLS.
  * Provided scoped server constructors and Layer helpers for easy application integration.
  * Added listen option types for non-TLS and TLS configurations.

* **Bug Fixes**
  * Improved accept-loop resilience with normalized open/accept errors, capped exponential backoff, and backoff reset after success.
  * Ensured connections and server fibers shut down cleanly, including interruption semantics and safe scope teardown.

* **Tests**
  * Added a Deno socket server test suite covering TCP/Unix echo, interruption, cleanup behavior, retry/backoff logic, and connection lifecycle.

### CODE REVIEW

File: packages/platform-deno/src/DenoSocketServer.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+import * as Layer from "effect/Layer"
+import type * as Scope from "effect/Scope"
+import * as SocketServer from "effect/unstable/socket/SocketServer"
+import { closeListener, fromListener } from "./internal/denoSocketServer.ts"
+
+type ListenOptions =
+  | (Deno.TcpListenOptions & { transport?: "tcp" })
+  | (Deno.UnixListenOptions & { transport: "unix" })
+
+type TlsListenOptions = Deno.ListenTlsOptions & Deno.TlsCertifiedKeyPem
```
Comment (tim-smart) [inline packages/platform-deno/src/DenoSocketServer.ts]: export these types
